src/thunderbots/software/network_input/filter/robot_filter.cpp

```diff
@@ -2,20 +2,72 @@
 
 #include <ros/console.h>
 
-RobotFilter::RobotFilter(unsigned int id) : robot_id(id) {}
+/**
+ * Implementation of the robot filter
+ *
+ * Helps with resolving robot position when there are mutiple data structs
+ * with the same robot id by averaging, and deals with missing data structs
+ * by caching
+ *
+ */
+
+RobotFilter::RobotFilter(unsigned int id) {}
 
 FilteredRobotData RobotFilter::getFilteredData(
-    const std::vector<SSLRobotData> &new_robot_data)
+    const std::vector<SSLRobotData>& new_robot_data)
 {
-    FilteredRobotData filtered_data;
-    filtered_data.id               = 0;
-    filtered_data.position         = Point();
-    filtered_data.velocity         = Vector();
-    filtered_data.orientation      = Angle::zero();
-    filtered_data.angular_velocity = AngularVelocity::zero();
-    filtered_data.timestamp        = Timestamp::fromSeconds(0);
-
-    return filtered_data;
+    // return cache if there is no data
+    if (new_robot_data.empty())
+    {
+        return cached_robot_data;
```

Comment:
What if we have no cached robot data? The two options I can see here are we can return an `std::optional`, or have the constructor for the filter take a robot state, so we can guarantee that we will always have at least one state.

Reply:
Can we just use a default constructor and init everything to zero? The data would be moving so fast that the initial data will probably not matter as much. It would make it a bit easier on the team filters end and remove logic for handling the first case?

But yeah the current approach just returns whatever garbage that `cached_robot_data` is holding on to, it needs to change

Reply:
I'm fine with initializing everything to 0 in the default constructor, as long as we're explicit in documenting this behavior everywhere that it is relevant. :+1: